### ISSUE TITLE
🐛 BUG: LIBC error when running pages dev

### ISSUE THREAD

**Issue by Gregoor** [cloudflare/workers-sdk#3788]
### Which Cloudflare product(s) does this pertain to?

Pages

### What version(s) of the tool(s) are you using?

3.5.1

### What version of Node are you using?

20.5.1

### What operating system are you using?

Mac

### Describe the Bug

When running `wrangler pages dev` I get

```
Automatically determined the proxy port to be 5174.
▲ [WARNING] No compatibility_date was specified. Using today's date: 2023-08-18.

  Pass it in your terminal:
  ```
  --compatibility-date=2023-08-18
  ```
  See https://developers.cloudflare.com/workers/platform/compatibility-dates/ for more information.


Compiling worker to "/var/folders/z6/6cmn96692tq6nfl1x56_32lh0000gn/T/functionsWorker-0.11404667805163804.mjs"...
✘ [ERROR] The constant "GLIBC" must be initialized

    functions/node_modules/.pnpm/detect-libc@2.0.2/node_modules/detect-libc/index.d.ts:4:13:
      4 │ export const GLIBC: 'glibc';
        ╵              ~~~~~

✘ [ERROR] The constant "MUSL" must be initialized

    functions/node_modules/.pnpm/detect-libc@2.0.2/node_modules/detect-libc/index.d.ts:5:13:
      5 │ export const MUSL: 'musl';
        ╵              ~~~~


✘ [ERROR] The constant "GLIBC" must be initialized

    functions/node_modules/.pnpm/detect-libc@2.0.2/node_modules/detect-libc/index.d.ts:4:13:
      4 │ export const GLIBC: 'glibc';
        ╵              ~~~~~


✘ [ERROR] The constant "MUSL" must be initialized

    functions/node_modules/.pnpm/detect-libc@2.0.2/node_modules/detect-libc/index.d.ts:5:13:
      5 │ export const MUSL: 'musl';
        ╵              ~~~~


✘ [ERROR] Build failed with 2 errors:

  functions/node_modules/.pnpm/detect-libc@2.0.2/node_modules/detect-libc/index.d.ts:4:13: ERROR:
  The constant "GLIBC" must be initialized
  functions/node_modules/.pnpm/detect-libc@2.0.2/node_modules/detect-libc/index.d.ts:5:13: ERROR:
  The constant "MUSL" must be initialized


 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by JacobMGEvans** [comment]
Greetings! We appreciate the time and effort to report this issue. To allow us to help you better please provide additional logging, information and if possible the most helpful, a [minimal viable reproduction.](https://stackoverflow.com/help/minimal-reproducible-example) 

Thanks! 😄

**Comment by Gregoor** [comment]
Fair! I've assumed it'd be sth obvious but it might not be. I have a repro of _something_ here:

https://github.com/Gregoor/wrangler-repro

where when I run `pnpm dev` I'm getting

```
✘ [ERROR] MiniflareCoreError [ERR_RUNTIME_FAILURE]: The Workers runtime failed to start. There is likely additional logging output above.
```

with no other error logged above. Though this may not be the same issue?

**Comment by Gregoor** [comment]
Okay my repro apparently does not repro correctly. Over there it works when I switch away from TypeScript and pnpm workspaces. Both of those are supported though right?

**Comment by penalosa** [comment]
@Gregoor I'm going to close this for now since I can't reproduce the the issue with your linked repository:

Running `pnpm i && pnpm dev` gives me:

```sh
Scope: all 2 workspace projects
Lockfile is up to date, resolution step is skipped
Already up to date
Done in 690ms

> wrangler-repro@1.0.0 dev /path/to/wrangler/wrangler-repro
> wrangler pages dev --compatibility-date=2023-08-14 .

Compiling worker to "/var/folders/wl/xdqbkld10134zqq2kp70lsgr0000gp/T/functionsWorker-0.8583379781337255.mjs"...
✨ Compiled Worker successfully
 ⛅️ wrangler 3.5.1 (update available 3.9.0)
-----------------------------------------------------
wrangler dev now uses local mode by default, powered by 🔥 Miniflare and 👷 workerd.
To run an edge preview session for your Worker, use wrangler dev --remote
⎔ Starting local server...
[mf:inf] Ready on http://127.0.0.1:8788/
[mf:inf] GET / 200 OK (34ms)
[mf:inf] GET /hi 200 OK (2ms)
[mf:inf] GET /favicon.ico 200 OK (3ms)
⎔ Shutting down local server...
```

Feel free to comment with more details if you're still running into issues, and we can investigate further!

**Comment by 1000hz** [comment]
It seems like the `libc` errors are coming from a dependency on `better-sqlite3` in miniflare, which has since been removed in more recent versions. We're gearing up to release `wrangler@3.9.1` very shortly which updates miniflare to one without the `better-sqlite3`. Be sure to give upgrading to this version a shot if you're still running into trouble, @Gregoor.
